torch/csrc/jit/frontend/schema_matching.cpp

```diff
@@ -284,6 +284,19 @@ static bool varargsCanBeUsedAsList(
       !typevar_list;
 }
 
+// Note (@zasdfgbnm):
+// This is a workaround for https://github.com/pytorch/pytorch/issues/47964
+// Currently JIT does not distinguish ScalarType vs int, so there is really
+// no way to distinguish x.view(1) vs x.view(torch.int8). So we have to hardcode
```

Comment:
Which overload is x.view(torch.int8)?

Reply:
Without this logic, both `x.view(1)` and `x.view(torch.int8)` get matched to `aten::view.dtype`. Even `x.view(-1)` get matched to `aten::view.dtype` although `-1` is not a valid dtype.

Actuacally I think `aten::view.dtype` gets priority over `aten::view` because `aten::view.dtype` can be matched at `tryMatchSchema(..., allow_conversions=False)`, but `aten::view` is usually matched by `tryMatchSchema(..., allow_conversions=True)`